File: java/toolgood.algorithm/src/main/java/toolgood/algorithm/OperandNull.java
package toolgood.algorithm;

class OperandNull extends Operand {
    // public override OperandType Type => OperandType.NULL;
    // public override bool IsNull => true;

    @Override
    public OperandType Type() {
        return OperandType.NULL;
    }

    @Override
    public boolean IsNull()   {
        return true;
    }

}

File: csharp/ToolGood.Algorithm/Operands/OperandNull.cs
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm
{
	internal sealed class OperandNull : Operand
	{
		public override bool IsNull => true;
		public override bool IsNotNull => false;
		public override OperandType Type => OperandType.NULL;
		public override string ToString() { return "null"; }
	}
}